### ISSUE TITLE
 ERROR in ./node_modules/countup.js-angular2/countUp.module.ts

### ISSUE THREAD

**Issue by Farre82** [inorganik/countUp.js-angular2#6]
Module build failed: Error: C:\XXX\node_modules\countup.js-angular2\countUp.directive.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include'
property.
    The missing file seems to be part of a third party library. TS files in published libraries are often a sign of a badly packaged library. Please open an issue in the library repository to alert its author and ask them to package the library using the Angular Package Format (https://goo.gl/jB3GVv).

**Comment by inorganik** [comment]
How are you including the countUp.js-angular2 module?

**Comment by Farre82** [comment]
Hi,
I' m using angular 5 and is including countup.js in my sharedModule like this.

`import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";
import { FormsModule } from "@angular/forms";
import { CountUpModule } from "countup.js-angular2";

@NgModule({
    imports: [
        CommonModule,
        FormsModule,
        CountUpModule
    ],
    declarations: [
    ],
    exports: [
        CommonModule,
        FormsModule,
        CountUpModule,
    ]
})
export class SharedModule { }
`

**Comment by Farre82** [comment]
Hi again,
I think it have to do with Angular 5 and AOT. 
See this link [https://github.com/angular/angular-cli/issues/8263](url)

**Comment by valdenio** [comment]
The problem appears when we try to compile for production (in my case it's in Ionic, but i think it's the same problem of the others). The Angular compiler (version 5.2.9 in my case) is looking for the compiled js files, but in the npm module there are only the typescript files.

**Comment by iplaksiy** [comment]
Having the same issue now that I upgraded to v6.

`ng build --prod`

```
ERROR in ./node_modules/countup.js-angular2/countUp.directive.ts
Module build failed: Error: /Users/iplaksiy/Development/Product-Engineering/retail-relevance-calculator/node_modules/countup.js-angular2/countUp.directive.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
The missing file seems to be part of a third party library. TS files in published libraries are often a sign of a badly packaged library. Please open an issue in the library repository to alert its author and ask them to package the library using the Angular Package Format (https://goo.gl/jB3GVv).
    at AngularCompilerPlugin.getCompiledFile (/Users/iplaksiy/Development/Product-Engineering/retail-relevance-calculator/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:708:23)
    at plugin.done.then (/Users/iplaksiy/Development/Product-Engineering/retail-relevance-calculator/node_modules/@ngtools/webpack/src/loader.js:41:31)
    at process._tickCallback (internal/process/next_tick.js:178:7)
ERROR in ./node_modules/countup.js-angular2/countUp.module.ts
Module build failed: Error: /Users/iplaksiy/Development/Product-Engineering/retail-relevance-calculator/node_modules/countup.js-angular2/countUp.module.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
The missing file seems to be part of a third party library. TS files in published libraries are often a sign of a badly packaged library. Please open an issue in the library repository to alert its author and ask them to package the library using the Angular Package Format (https://goo.gl/jB3GVv).
    at AngularCompilerPlugin.getCompiledFile (/Users/iplaksiy/Development/Product-Engineering/retail-relevance-calculator/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:708:23)
    at plugin.done.then (/Users/iplaksiy/Development/Product-Engineering/retail-relevance-calculator/node_modules/@ngtools/webpack/src/loader.js:41:31)
    at process._tickCallback (internal/process/next_tick.js:178:7)
```

Angular **6.0.0**
Angular CLI: **6.0.0**
CountUp: **1.1.1**

It was building fine in v5.*, but it looks like with the upgrade, that's no longer the case.

**Comment by mjacobson7** [comment]
Any word on this?  I am experiencing this as well.  As @iplaksiy noted, the error message says the following:

`Please open an issue in the library repository to alert its author and ask them to package the library using the Angular Package Format (https://goo.gl/jB3GVv). `

**Comment by inorganik** [comment]
For now, use @iplaksiy's solution. I tried rerunning a newer version of ng-packagr, and it results in an AOT error: 
```
ERROR in : Unexpected value 'CountUpModule in /.../node_modules/countup.js-angular2/countUp.module.d.ts' imported by the module 'AppModule in /.../src/app/app.module.ts'. Please add a @NgModule annotation.
```
Angular still hasn't figured out a good way to pass directives around via npm. Of course, you can always just copy the count-up directive in this repo and declare it in app.module.ts. It's really that simple! But due to all the complicated webpack bundling, transpiling, etc, we have to jump through so many hoops if we want to just npm install it. * sigh *

**Comment by inorganik** [comment]
Fixed in v6.0.0
